### ISSUE TITLE
Make RD settings/preferences applicable globally or by project

### ISSUE THREAD

**Issue by ThunderFrame** [rubberduck-vba/Rubberduck#940]
Might be useful to be able to have RD settings that are applicable to a particular project.

For example, you might have inherited a particularly nasty project that has lot's of Warnings. You don't want to refactor the whole project, but nor do you want to have to change your RD settings back and forth every time you open and close the project.

I suppose this could be implemented as Commented settings in a module, or as VB_Ext attributes in a module.

I guess an alternative to settings at the project level would be an ability to load/save pre-configured settings. Eg. MyStandardSettings.xml, MyTolerantSettings.xml, GiveMeStrengthSettings.xml etc.

**Comment by Hosch250** [comment]
I don't really see why this would necessary.  With Source Control and the auto-fixes most inspections have, this would be a breeze to do and control.

**Comment by retailcoder** [comment]
@Hosch250 I'm not sure I see how source control and inspections/quickfixes apply to this feature request.. settings aren't under source control, and they're loaded from a somewhat hard-coded path & filename. the idea would be to allow loading different settings for different project, like, inspections XYZ could be warnings in one project, be disabled in another, and show up as hints in another.

I don't like cluttering users' code with comments though, `VB_Ext` attributes would be ideal, but there's no .vbp project file to put them in, and we wouldn't want them in a random module, nor in a preset/dedicated module either - the best solution seems to simply allow the user to load any arbitrary rubberduck configuration file - i.e. use the hard-coded path for default settings, and store the path of the config file to load and apply (or null/empty string if no config file should be loaded).

**Comment by Hosch250** [comment]
@retailcoder My point is that most or all of the inspections shouldn't be at one level in one project and a different level in another.  Granted, there are instances in which the warnings are about desired behavior, but this is unlikely to be project-wide.  I think refactoring the codebase would be better here - especially since the "refactoring" is mostly "small" things like explicitly stating the access modifier and such.

**Comment by retailcoder** [comment]
I can't disagree with that :+1:

**Comment by ckuhn203** [comment]
I can disagree with that. I actually do use different R# settings for different projects at work. Say you've got a legacy code base X where Inspection x' is just too noisy and can't be fixed right now. I'd like to hush x' up, but only in legacy code base X. Over in greenfield project Y, I want inspection x' to show up. 

I think that's the core of this feature request, and it's a good one, even if it's a "nice to have" feature.

**Comment by ckuhn203** [comment]
TL;DR: :+1: , but I'm not sure how we can reliably implement *project* settings. I think allowing the user to export/load arbitrary settings files at runtime would be acceptable though.

**Comment by Vogel612** [comment]
closing as forward duplicate of the project file idea (which in turn was closed as retriage-v3)
